mlrun/feature_store/feature_set.py

```diff
@@ -1277,18 +1291,76 @@ def do(self, event):
                 for window in windows:
                     spark_window = self._duration_to_spark_format(window)
                     aggs = last_value_aggs
+                    all_aggs_names = ""
                     for operation in operations:
                         agg = self._get_aggr(operation, column)
                         agg_name = f"{name if name else column}_{operation}_{window}"
                         agg = agg.alias(agg_name)
                         aggs.append(agg)
+                        all_aggs_names = all_aggs_names + agg_name
                     window_column = funcs.window(
                         time_column, spark_window, spark_period
                     )
-                    df = input_df.groupBy(
-                        *self.key_columns,
-                        window_column.end.alias(time_column),
-                    ).agg(*aggs)
+                    if input_df.isStreaming:
+
+                        class MyBatchProcessor:
+                            def __init__(self):
+                                self.df = None
+
+                            def __call__(self, df, epoch_id):
+                                self.df = df
+
+                        batchProcessor = MyBatchProcessor()
+
+                        df = (
+                            input_df.withWatermark(time_column, spark_window)
+                            .groupBy(
+                                *self.key_columns,
+                                window_column.alias(time_column),
+                            )
+                            .agg(*aggs)
+                        )
+
+                        if self.checkpoint_path:
+                            checkpoint_path = (
+                                self.checkpoint_path
+                                if self.checkpoint_path.endswith("/")
+                                else self.checkpoint_path + "/"
+                            )
```

Comment:
This is shorter and probably more readable:
```suggestion
                            checkpoint_path = self.checkpoint_path
                            if not checkpoint_path.endswith("/"):
                                checkpoint_path += "/"
```